### ISSUE TITLE
After fresh installation Flarum fails to work with error

### ISSUE THREAD

**Issue by Pilvinen** [flarum/core#1731]
## Bug Report

**Current Behavior**
After fresh installation Flarum fails to work.

**Steps to Reproduce**
1. Install fresh flarum by following the installation instructions.
2. After installation is done Flarum fails to work with error message: https://pastebin.com/qaigk7Mj

**Expected Behavior**
Should work.

**Environment**
- Flarum version: 0.1.0-beta.8.1
- Website URL: https://www.ritualsoftheold.com/community-test/public/
- Webserver: Apache
- Hosting environment: VPS
- PHP version: 7.2.10-0ubuntu0.18.04.1
- Browser: Google Chrome 71.0.3578.98

```
Flarum core 0.1.0-beta.8.1
PHP version: 7.2.10-0ubuntu0.18.04.1
Loaded extensions: Core, date, libxml, openssl, pcre, zlib, filter, hash, pcntl, Reflection, SPL, sodium, session, standard, mysqlnd, PDO, xml, apcu, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, imagick, json, exif, mysqli, pdo_mysql, Phar, posix, readline, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, Zend OPcache
+----------------------+-----------------+--------+
| Flarum Extensions    |                 |        |
+----------------------+-----------------+--------+
| ID                   | Version         | Commit |
+----------------------+-----------------+--------+
| flarum-approval      | v0.1.0-beta.8   |        |
| flarum-bbcode        | v0.1.0-beta.8   |        |
| flarum-emoji         | v0.1.0-beta.8   |        |
| flarum-lang-english  | v0.1.0-beta.8   |        |
| flarum-flags         | v0.1.0-beta.8.1 |        |
| flarum-likes         | v0.1.0-beta.8.1 |        |
| flarum-lock          | v0.1.0-beta.8   |        |
| flarum-markdown      | v0.1.0-beta.8   |        |
| flarum-mentions      | v0.1.0-beta.8.1 |        |
| flarum-statistics    | v0.1.0-beta.8   |        |
| flarum-sticky        | v0.1.0-beta.8   |        |
| flarum-subscriptions | v0.1.0-beta.8   |        |
| flarum-suspend       | v0.1.0-beta.8   |        |
| flarum-tags          | v0.1.0-beta.8.2 |        |
+----------------------+-----------------+--------+
Base URL: https://www.ritualsoftheold.com/community-test/public
Installation path: /var/www/ritualsoftheold/public_html/community-test
Debug mode: ON
Don't forget to turn off debug mode! It should never be turned on in a production system.
```

**Comment by luceos** [comment]
Try running `php flarum cache:clear` and/or `rm -rf storage/cache/textformatter/*` and please let me know whether that works.

**Comment by Pilvinen** [comment]
No change.

**Comment by nartamus** [comment]
If you can re-image the VPS and start over, you can try this script I wrote for myself.
`https://github.com/nartamus/flarum-install-scripts`

Don't install anything, just download the script and run `sudo bash flarum-install.sh` 
I have tested this on Ubuntu 14,16 and 18. You can change the domain, email and db password in the .sh file.

**Comment by JoshyPHP** [comment]
I think I've seen this before, some version of PHP where `DOMDocument::$documentElement` is null. Try upgrading your libxml2 if possible. I'm not sure whether the issue is in PHP, libxml2 or if it's some sort of ABI issue with mismatched versions, so using another version of PHP or libxml2 may fix it.

**Comment by Pilvinen** [comment]
I checked out your script.

1) It installs Mariadb - not gonna do that.
2) It installs old 7.1 versions of php etc - not gonna do that either.
3) It installs Flarum in the usual way - already did that.
4) It creates a virtual host file for a domain - I run my Flarum from a subdirectory.
5) It applies permissions - I already tested a+rwx to everything which proved it's not a permissions related issue.
6) It creates a database and grants privileges - I'm already doing that.

**To sum it up:**

It's a great script and could potentially help a great many people, but from my analysis I don't see how it would address the _Flarum bug_.

**Comment by Pilvinen** [comment]
Libxml2:
`libxml2 is already the newest version (2.9.4+dfsg1-6.1ubuntu1.2)`

Php:
`PHP 7.2.10-0ubuntu0.18.04.1 (cli) (built: Sep 13 2018 13:45:02) ( NTS )`

**Comment by JoshyPHP** [comment]
It doesn't really matter whether it's a newer version or an older one, you're just looking for a combination that works. There's a chance that manually compiling a vanilla version of PHP with the default extensions would work too.

The core issue is that there's a bit missing from your version of PHP and replacing it may fix it.

**Comment by Pilvinen** [comment]
Well, let me know if you need any more information to help you fix this **Flarum bug**. I'll be happy to help you.

**Comment by franzliedke** [comment]
@Pilvinen Whether this is a Flarum bug or not hasn't been established yet, as far as I can tell. To the best of my knowledge, there haven't been any similar reports in the latest version of Flarum.

Since you are asking what you can do... Please investigate whether @JoshyPHP's (who is awesome and maintains the wonderful TextFormatter library) intuition is correct, and whether another combination of PHP and libxml or even recompiling PHP helps.

**Comment by Pilvinen** [comment]
I use the libraries that came out of the box with Ubuntu server 18.04.

**Comment by nartamus** [comment]
[Video Showing Install](https://www.youtube.com/watch?v=6rSd3PgZeYs&)

Above is a link of me installing Flarum on Ubuntu 18.04, MySQL and PHP7.2. Those are your specs, and the process should be the same for both of us. Maybe you can see something different that I'm doing that you're not.

**Comment by jordanjay29** [comment]
This doesn't appear to be a bug but a server configuration issue, so I'm closing this. Please seek installation support on our forums at discuss.flarum.org.
